code/modules/mob/living/carbon/xenomorph/evolution.dm

```diff
@@ -196,6 +195,15 @@
 			to_chat(src, "<span class='warning'>Something in this place is interfering with our link to Queen Mother. We are unable to evolve to a psychic caste here!</span>")
 			return
 
+	else if(new_caste_type == /mob/living/carbon/xenomorph/hivemind) //Special case for dealing with hiveminds - this may be subject to heavy change, such as multiple hiveminds potentially being an option
+		if(length(hive.xenos_by_typepath[/mob/living/carbon/xenomorph/hivemind]))
+			to_chat(src, "<span class='warning'>There cannot be two manifestations of the hivemind's will at once..</span>")
+			return
+
+		if(isxenoresearcharea(get_area(src)))
+			to_chat(src, "<span class='warning'>Something in this place is interfering with our link to the Hivemind. We are unable to evolve to be its manifestation!</span>")
+			return
```

Comment:
You still need to add a check to stop someone creating a hiveminnd in a spot that could get them stuck or allow abuse.

Reply:
hiveminds cant exactly 'move', and in the case of getting their camera stuck ,they can doubleclick weeds to jump to them, or just jump to a xeno directly.

Ill still probably add a check eventually to see if it can be placed